code/modules/mob/living/simple_animal/hostile/hivebot.dm

```diff
@@ -31,85 +68,207 @@
-	..()
-	visible_message("<b>[src]</b> blows apart!")
-	new /obj/effect/decal/cleanable/blood/gibs/robot(src.loc)
-	spark(src, 3, alldirs)
+	..(null,"blows apart!")
+	if(linked_parent)
+		linked_parent.existing_bots -= 1
+		linked_parent.linked_bots -= src
+	var/T = get_turf(src)
+	new /obj/effect/gibspawner/robot(T)
+	spark(T, 1, alldirs)
 	qdel(src)
 	return
 
-/mob/living/simple_animal/hostile/hivebot/tele//this still needs work
-	name = "Beacon"
-	desc = "Some odd beacon thing"
-	icon_state = "def_radar-off"
-	icon_living = "def_radar-off"
+/mob/living/simple_animal/hostile/hivebot/think()
+	. =..()
+	if(stance == HOSTILE_STANCE_IDLE)
+		icon_state = "[initial(icon_state)]"
+	else
+		icon_state = "[initial(icon_state)]_armed"
+
+/mob/living/simple_animal/hostile/hivebot/proc/wakeup()
+	stance = HOSTILE_STANCE_IDLE
+
+/mob/living/simple_animal/hostile/hivebot/emp_act(severity)
+	LoseTarget()
+	stance = HOSTILE_STANCE_TIRED
+	addtimer(CALLBACK(src, .proc/wakeup), 150)
+	if(severity == 1.0)
+		apply_damage(10)
+
+/mob/living/simple_animal/hostile/hivebotbeacon
+	name = "Hivebot beacon"
+	desc = "An odd and primitive looking machine. It emanates of strange and powerful energies. It bears no manufacturer markings of any kind."
+	icon = 'icons/mob/npc/hivebot.dmi'
+	icon_state = "hivebotbeacon_active"
+	icon_living = "hivebotbeacon_active"
 	health = 200
 	maxHealth = 200
+	projectilesound = 'sound/weapons/taser2.ogg'
+	projectiletype = /obj/item/projectile/beam/hivebotdischarge
+	wander = 0
 	status_flags = 0
-	anchored = 1
-	stop_automated_movement = 1
-	var/bot_type = NORMAL // type of bot, 1 is normal, 2 is ranged, 3 is rapid rnaged.
-	var/bot_amt = 10
-	var/spawn_delay = 600
+	faction = "hivebot"
+	ranged = 1
+	rapid = 1
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	minbodytemp = 0
+	speed = -10
+	see_in_dark = 8
+	smart = 1
+	destroy_surroundings = 0
+	var/bot_type = NORMAL // type of bot, 1 is normal, 2 is ranged, 3 is rapid ranged
+	var/bot_amt = 48 //Number of total bots that are spawned before the beacon disappears completely.
+	var/max_bots = 16 //Number of bots linked to this beacon specifically that can exist, before spawning more is halted.
+	var/list/linked_bots = list()
+	var/latest_child = null
+	var/existing_bots = 0
+	var/spawn_delay = 300
+	var/activated = 0
+	var/snoozing = 0 //If set to 1, it will be prevented from spawning bots, unless it spots an enemy.
+	attack_emote = "focuses on"
+
+/mob/living/simple_animal/hostile/hivebotbeacon/toxic
+	projectiletype = /obj/item/projectile/beam/hivebotbeam
+	projectilesound = 'sound/weapons/laser3.ogg'
+	rapid = 0
 
-/mob/living/simple_animal/hostile/hivebot/tele/New()
+/mob/living/simple_animal/hostile/hivebotbeacon/incendiary
+	projectiletype = /obj/item/projectile/beam/hivebotincendiary
+	projectilesound = 'sound/weapons/plasma_cutter.ogg'
+	rapid = 0
+
+/mob/living/simple_animal/hostile/hivebotbeacon/New()

```

Comment:
```suggestion
/mob/living/simple_animal/hostile/hivebotbeacon/Initialize(mapload)
```

Might want to convert this to Initialize. It is preferred, specially with the sparking. If done, you also need to change the `..()` to `. = ..()` and you might want to guard the sparking against mapload (since having sparks fly up when the map is loaded is kinda pointless).

```dm
/mob/living/simple_animal/hostile/hivebotbeacon/Initialize(mapload)
	. = ..()
	if (!mapload)
		var/datum/effect/effect/system/smoke_spread/S = new /datum/effect/effect/system/smoke_spread()
		S.set_up(5, 0, src.loc)
		S.start()

	icon_state = "hivebotbeacon_off"
	visible_message("<span class='danger'>\ [src] warps in!</span>")
	playsound(src.loc, 'sound/effects/EMPulse.ogg', 25, 1)
```